llvm/utils/UpdateTestChecks/common.py

```diff
@@ -340,9 +349,57 @@ def get_checks_for_unused_prefixes(
         return ret
 
 
+def parse_lit_substitutions(
+    test_path: str, test_suite_cache, local_config_cache
+) -> list:
+    """Reason about lit config files and derive which substitutions are to be
+    used for the test at test_path."""
+
+    # Make sure to import lit from llvm/utils
+    utils_dir = os.path.dirname(os.path.dirname(__file__))
+    lit_dir = os.path.join(utils_dir, "lit")
+    if not lit_dir in sys.path:
+        sys.path.insert(0, lit_dir)
+
+    import lit.LitConfig
+    import lit.discovery
+
+    # The important bit here is allow_partial_configs. We won't be able to
+    # understand the whole tree of config files (including lit.site.cfg) without
+    # access to a build tree. So just accept some bits won't be loaded. This
+    # is more that enough to understand lit.local.cfg files which mostly contain
+    # substitutions.
+    lit_config = lit.LitConfig.LitConfig(
+        progname="lit",
+        path=[],
+        quiet=False,
+        useValgrind=False,
+        valgrindLeakCheck=False,
+        valgrindArgs=[],
+        noExecute=False,
+        debug=_verbose,
+        isWindows=False,
+        order="smart",
+        params={},
+        allow_partial_configs=True,
+    )
+    lit_config.targets = []
+
+    tests = lit.discovery.getTests(
+        test_path, lit_config, test_suite_cache, local_config_cache
+    )[1]
+    try:
+        return next(tests).config.substitutions
+    except StopIteration:
+        return []
+
```

Comment:
This would be more readable as a `for`-loop. One of the weird quirks of Python is that for loops are just try-catch wrappers that look for the `StopIteration` exception.

Reply:
The thing is, I'm only interested in one of the tests, because I want to access the configuration for them, and this is shared across all. Ultimately, i'd like something like `tests[0]` or `tests.front()` but I don't think python provides this :/

Reply:
I guess I could write the following:
```
    for test in tests:
        return test.config.substitutions
    return []
```

I'm undecided whether it's clearer. What do you think?

Reply:
I think that you can also write something line `list(tests)[0]` which exhausts the iterator and then gets the first element. 

I agree with that it is a difficult case and I am not sure if there is a strictly better one. The for loop feels nicer to me since it doesn't have the try catch mechanism, which takes more knowledge to understand (ie StopIteration is not a fault).